src/Analyzers/CSharp/Analyzers/SimplifyLinqExpression/CSharpSimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,179 @@
+            }
+        }
+
+        public void AnalyzeAction(OperationAnalysisContext context, ImmutableHashSet<IMethodSymbol> whereMethods, ImmutableHashSet<IMethodSymbol> linqMethods)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var semanticModel = context.Operation.SemanticModel;
+            var invocationArgument = invocationOperation.Arguments.FirstOrDefault();
+
+            // Check to make sure the invocation argument is an InvocationExpressionSyntax
+            // Example: If invocationOperation syntax is Data().Where(...).Single(), then invocationArgument is Data.Where(...)
+            if (invocationArgument is null)
+            {
+                return;
+            }
+
+            var targetOperation = invocationArgument.Children.FirstOrDefault(c => c is IOperation);
+
+            // Example: If invocation is Data.Where(...), then the TargetMethod would be .Where(...)
+            if (targetOperation is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            var targetDefinition = invocation.TargetMethod.OriginalDefinition;
+
+            // True if the invocation.TargetMethod is a call to System.Linq.Enumerable.Where(...)
+            if (!whereMethods.Contains(targetDefinition))
+            {
+                return;
+            }
+
+            // True if the arguments within the .Where(...) method are lambda expressions
+            var isLambda = invocation.Arguments
+                .Any(c => (c.Syntax is ArgumentSyntax argSyntax && argSyntax.Expression.IsAnyLambda()) ||
+                           c.Syntax is LambdaExpressionSyntax);
+
+            if (!isLambda)
+            {
+                return;
+            }
+
+            // check that the Where clause is followed by a call to a valid method i.e. one of First, FirstOrDefault, Single, SingleOrDefault, etc..
+            // Example: if Data.Where(...).Single(), then the invocationOperation.TargetMethod is Single
+            var invocationDefinition = invocationOperation.TargetMethod.OriginalDefinition;
+
+            // True if invocationDefinition is one of:
+            // First(), Last(), Sinlge(), Any(), Count(), FirstOrDefualt(), LastOrDefault(), or SingleOrDefault()
+            if (!linqMethods.Contains(invocationDefinition))
+            {
+                return;
+            }
+
+            // Check that the Where clause is followed by a call with no predicate
+            var arguments = invocationOperation.TargetMethod.Parameters;
+            if (arguments.IsEmpty)
+            {
+                return;
+            }
+
+            var node = context.Operation.Syntax;
+
+            // Example: if Data().Where(...).First()
+            var targetMethodNode = node.DescendantNodes().OfType<IdentifierNameSyntax>().LastOrDefault();
+            var invokedNodeSyntax = targetOperation.Children.FirstOrDefault().Syntax;
```

Comment:
FirstOrDefault could return null.  this would crash.  i don't know what'd going on here.